docs/docs/guides/actions-events.md

```diff
@@ -1 +1,223 @@
+import { goBack } from "@rise-tools/kit-expo-router"
+import { haptics } from "@rise-tools/kit-haptics"
+
+function UI() {
+  return (
+    <Button
+      onPress={[goBack(), haptics()]}
+    >
+      Go back
+    </Button>
+  )
+}
+```
+
+## Events
+
+### Executing function on the server
+
+```tsx
+function UI() {
+  return (
+    <Button
+      onPress={() => console.log('This always runs on the server')}
+    >
+      Go back
+    </Button>
+  )
+}
+```
+
+Note: the above example is equivalent to:
+
+```tsx
+function UI() {
+  return (
+    <Button
+      onPress={event(() => console.log('This always runs on the server'))}
+    >
+      Go back
+    </Button>
+  )
+}
+```
+
+When you pass a function as a prop, we wrap it with `event()` automatically.
+
+### Passing additional options
+
+```tsx
+function UI() {
+  return (
+    <Button
+      onPress={event(() => /* redacted */, { timeout: 5000 })}
+    >
+      Go back
+    </Button>
+  )
+}
+```
+
+### Executing an action and a function on the server
+
+The following example will execute remote function on the server and execute provided actions, without waiting for the server response:
+
+```tsx
+import { goBack } from "@rise-tools/kit-expo-router"
+
+function UI() {
+  return (
+    <Button
+      onPress={event(() => /* redacted */, { actions: [goBack()] })}
+    >
+      Go back
+    </Button>
+  )
+}
+```
+
+## Responding to events
+
+### Return payload
+
+You can return an arbitrary JSON value from an event handler. This value will be sent back to the client. From client component, you can await the callback to read the response from the server. 
+
+<table>
+<thead>
+<tr>
+<th>Server</th>
+<th>Client</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>
+
+```tsx
+function UI() {
+  return (
+    <Form
+      onSubmit={() => {
+        return { message: 'Thank you!' }
+      }}
+    />
+  )
+}
+```
+
+</td>
+<td>
+
+```tsx
+function Form({ onSubmit }) {
+  const [isPending, setIsPending] = useState(false)
+  const handleSubmit = async () => {
+    setIsPending(true)
+    const { message } = await onSubmit()
+    setIsPending(false)
+    alert(message)
+  }
+  return (
+    <Button onPress={handleSubmit} disabled={isPending}>
+      {isPending ? <ActivityIndicator /> : "Submit"}
+    </Button>
+  )
+}
+```
+
+</td>
+</tr>
+</tbody>
+</table>
+
+### Execute an action
+
+You can also execute an action in response to an event:
+
+```tsx
+import { goBack } from "@rise-tools/kit-expo-router"
+
+function UI() {
+  return (
+    <Form
+      onSubmit={() => {
+        return goBack()
+      }}
+    />
+  )
+}
+```
+
+This will navigate back after the form is submitted and sucesfully processed by the server.
+
+### Execute multiple actions
+
+You can also execute multiple actions in response to an event:
+
+```tsx
+import { goBack } from "@rise-tools/kit-expo-router"
+import { haptics } from "@rise-tools/kit-haptics"
+
+function UI() {
+  return (
+    <Form
+      onSubmit={() => {
+        return [
```

Comment:
this behavior might be confusing. it looks like this would go back to the client with these events as the response.

I think we may require you to use `return response(something, { actions: [] })` here

Reply:
Yeah, I definitely agree, it might be confusing when both appear at the same time especially:
```ts
onSubmit={() => {
   if (success) return [action, action]
   return [{ error: 'Wrong name', field: 'name' }]
}
```

The challenge is that many times, you'll be just returning an action or multiple actions (e.g. in response to button press) and not returning any response:
```
return response(null, { actions: [] })
```

Feels like we could do better than that.

Do you have any ideas?

Reply:
idk, `responseActions([...])`

Reply:
not sure either. Let's keep this PR open for some time until we figure out a better API later today